cmd/rekor-cli/app/upload.go

```diff
@@ -141,6 +141,16 @@ var uploadCmd = &cobra.Command{
 		if err := verify.VerifySignedEntryTimestamp(ctx, &logEntry, verifier); err != nil {
 			return nil, fmt.Errorf("unable to verify entry was added to log: %w", err)
 		}
+		if logEntry.Verification.InclusionProof != nil {
+			// verify inclusion proof
+			if err := verify.VerifyInclusion(ctx, &logEntry); err != nil {
+				return nil, fmt.Errorf("error verifying inclusion proof: %w", err)
+			}
+			// verify checkpoint
+			if err := verify.VerifyCheckpointSignature(&logEntry, verifier); err != nil {
```

Comment:
what if checkpoint is nil? from old versions.

Reply:
In this upload case, we weren’t returning an inclusion proof previously. So this case handles both, since the checkpoint is a part of the inclusion proof now.

Reply:
Argh! I see, all of it's added into `createLogEntry` and we already return if we have a pre-existing entry here. gotcha, thx

Reply:
Yea, exactly. For the other cases, we returned an inclusion, but just not the checkpoint.